test/throttle_SUITE.erl

```diff
@@ -123,12 +128,50 @@ send_and_wait(_) ->
     ?assertMatch(ok, amoc_throttle:send_and_wait(?FUNCTION_NAME, receive_this)),
     %% One message is received sufficiently fast
     amoc_throttle:send(?FUNCTION_NAME, receive_this),
-    ?assertMatch(ok, receive_msg_in_timeout(receive_this, 100)),
+    ?assertMatch(ok, ?RECV(receive_this, 100)),
     %% If someone else fills the throttle heavily,
     %% it will take proportionally so long to execute for me
     fill_throttle(?FUNCTION_NAME, 100 * 10),
     amoc_throttle:send(?FUNCTION_NAME, receive_this),
-    ?assertMatch({error, not_received_yet}, receive_msg_in_timeout(receive_this, 200)).
+    ?assertMatch({error, not_received_yet}, ?RECV(receive_this, 200)).
+
+just_wait(_) ->
+    %% it failts if the throttle wasn't started yet
+    ?assertMatch({error, no_throttle_process_registered},
+                 amoc_throttle:wait(?FUNCTION_NAME)),
+    %% Start 100-per-10ms throttle with a single process
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100, 10, 1)),
+    %% wait passes fine
+    ?assertMatch(ok, amoc_throttle:wait(?FUNCTION_NAME)),
+    %% One message is received sufficiently fast
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch(ok, ?RECV(receive_this, 100)),
+    %% If someone else fills the throttle heavily,
+    %% it will take proportionally so long to execute for me
+    fill_throttle(?FUNCTION_NAME, 100 * 10),
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch({error, not_received_yet}, ?RECV(receive_this, 200)).
+
+wait_for_process_to_die_sends_a_kill(_) ->
+    erlang:process_flag(trap_exit, true),
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100, 10, 1)),
+    amoc_throttle:run(?FUNCTION_NAME, fun() -> exit(?FUNCTION_NAME) end),
+    ?assertMatch(ok, ?RECV({'EXIT', _, ?FUNCTION_NAME}, 100)).
+
+async_runner_dies_while_waiting_raises_exit(_) ->
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 1, 1, 1)),
+    find_new_link_and_kill_it(self()),
+    ?assertExit({throttle_wait_died, _, killed}, amoc_throttle:wait(?FUNCTION_NAME)).
+
+async_runner_dies_when_throttler_dies(_) ->
+    erlang:process_flag(trap_exit, true),
+    {links, OriginalLinks} = erlang:process_info(self(), links),
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 1, 60000, 1)),
+    wait_until_one_throttle_worker(?FUNCTION_NAME),
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    wait_until_one_async_runner(self(), OriginalLinks),
+    kill_throttle_workers(?FUNCTION_NAME),
```

Comment:
we can just stop the trottle with amoc_trottle:stop/1

Reply:
Oh, true :D